src/compiler/factory.ts

```diff
@@ -1207,6 +1207,54 @@ namespace ts {
             : node;
     }
 
+    export function createTemplateHead(text: string) {
+        const node = <TemplateHead>createSynthesizedNode(SyntaxKind.TemplateHead);
+        node.text = text;
+        return node;
+    }
+
+    export function updateTemplateHead(node: TemplateHead, text: string) {

```

Comment:
Generally we only update `Node` or `NodeArray` properties, not strings (e.g. there's no update method for `Identifier`). I don't think we need these update methods.

Reply:
Alright then, will remove.